proto/zitadel/user/schema/v1alpha/user_schema.proto

```diff
@@ -0,0 +1,117 @@
+syntax = "proto3";
+
+package zitadel.user.schema.v1alpha;
+
+import "zitadel/object/v2beta/object.proto";
+import "google/protobuf/struct.proto";
+import "validate/validate.proto";
+import "protoc-gen-openapiv2/options/annotations.proto";
+
+option go_package = "github.com/zitadel/zitadel/pkg/grpc/user/schema/v1alpha";
+
+message UserSchema {
+
+  // ID is the read-only unique identifier of the schema.
+  string id = 1 [
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
+      example: "\"69629012906488334\""
+    }
+  ];
+  // Details provide some base information (such as the last change date) of the schema.
+  zitadel.object.v2beta.Details details = 2;
+  // Type is a human readable text describing the schema.
+  string type = 3 [
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
+      example: "\"employees\""
+    }
+  ];
+  // Current state of the schema.
+  State state = 4 [
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
+      example: "\"STATE_ACTIVE\""
+    }
+  ];
+  // Revision is a read only version of the schema, each update increases the revision.
+  uint32 revision = 5 [
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
+      example: "\"2\""
+    }
+  ];
+  // JSON schema representation defining the user.
+  google.protobuf.Struct schema = 6 [
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
+      example: "{\"$schema\":\"https://example.com/user/employees\",\"type\":\"object\",\"properties\":{\"name\":{\"type\":\"string\",\"required\":true},\"description\":{\"type\":\"string\"}}}"
+    }
+  ];
+  // Defines the possible types of authenticators.
+  // This allows creating different user types like human/machine without usage of actions to validate possible authenticators.
+  // Removal of an authenticator does not remove the authenticator on a user.
+  repeated AuthenticatorType possible_authenticators = 7  [
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
+      example: "[\"AUTHENTICATOR_TYPE_USERNAME\",\"AUTHENTICATOR_TYPE_PASSWORD\",\"AUTHENTICATOR_TYPE_WEBAUTHN\"]";
+    }
+  ];
+}
+
+enum FieldName {
+  FIELD_NAME_UNSPECIFIED = 0;
+  FIELD_NAME_TYPE = 1;
+  FIELD_NAME_STATE = 2;
+  FIELD_NAME_REVISION = 3;
+  FIELD_NAME_CREATION_DATE = 4;
+}
+
+message SearchQuery {
+  oneof query {
+    option (validate.required) = true;
+
+    // Limit the result to a specific schema type.
+    TypeQuery type_query = 1;
+    // Limit the result to a specific state of the schema.
+    StateQuery state_query = 2;
+  }
+}
```

Comment:
do we want to include (and, or , not) here as well?

Reply:
Is there no way to query a defined schema field? I mean you could theoretically check with the schema if a field like username is provided right?

Reply:
You mean kind of:

> Get me all schema, where at any point there is a field `firstname` defined.

do we really need that? at least right away?